### ISSUE TITLE
Footnotes containing Greek can cause problems

### ISSUE THREAD

**Issue by wf49670** [davem2/dp2ppgen#4]
Running with -pcfjse --fndest=chapterend and using 0.2.0.

Input:

> no to me, what about my own duty to him? Seeing that a
> 
> [Footnote 1: [Greek:
> 
> /*
>          pylla[**polla? P1] d' en metaichmiô
> notos kylindei kymat' eureiês halos.
> */
> 
> ]
> 
> The author is unknown.]

Output: 

> no to me, what about my own duty to him? Seeing that a
> 
> The author is unknown.]

The footnote is generated at the end of the chapter, but includes only the text up to the ] that ends the Greek tag.

I am not sure this formatting markup is correct, though. More likely the [Greek: tag should have started inside the /\* */ block. However, that does not help, as shown in this subsequent example:

> [Footnote 1: Homer, <i>Od.</i>.xii. 209:
> 
> /*
> [Greek: on men dê tode meixon epi kakon ê ote Kuklôps
> eilei eni skêi glaphyrps kraterêphi biêphi.]
> */
> 
> Cicero means that they have no more reason to resist Cæsar than before,
> for his claim to keep his province is not more unconstitutional than his
> former claims. "We can't suppose that our object in yielding the old
> demands was to arm him against ourselves!" <i>Nisi forte</i> is here practically
> a negative; it introduces an impossible supposition.]

For that one, the moved footnote also ended at the ] ending the Greek, leaving the remaining lines beginning with the */ at the bottom of the original page.

**By the way**, the help for dp2ppgen indicates that --fndest an be "inline", but if one specifies that all the footnotes seem to disappear. 

[**Edit:** I was hoping it would just leave them where encountered, though rejoining across page boundaries. That's the option I would ultimately like for working with the enhanced footnote processing in ppgen when footnotes are to occur at chapter- or book-end. Leave them where you found them, but emit the .fm directives at each chapter end, or at the book end. Then the PPer can adjust the .fm options as desired. The only time you'd need to move a footnote would be when the PPer wants them after paragraphs.]

**Comment by davem2** [comment]
Thanks for the report Walt,

I fixed the issue with [Greek] in footnotes late last night. Could you update to the latest version and try again? I'm not seeing the issues you are using the latest version. 

Also, for the time being until things are more complete, I'm only bumping the version on tagged releases. To reference a build it would be more helpful to provide the commit hash instead of the version.

> By the way, the help for dp2ppgen indicates that --fndest an be "inline", but if one specifies that all the footnotes seem to disappear. 

Inline footnotes have not been implemented yet, but I probably should put a warning or something up when fndest==inline for the time being. 

> [Edit: I was hoping it would just leave them where encountered, though rejoining across page boundaries. That's the option I would ultimately like for working with the enhanced footnote processing in ppgen when footnotes are to occur at chapter- or book-end. Leave them where you found them, but emit the .fm directives at each chapter end, or at the book end. Then the PPer can adjust the .fm options as desired. The only time you'd need to move a footnote would be when the PPer wants them after paragraphs.] 

Currently, dp2ppgen still handles footnotes according to the old footnote ppgen implementation. Footnotes will work the way you're describing once I've updated dp2ppgen.

**Comment by wf49670** [comment]
On 3/17/2015 3:51 PM, davem2 wrote:

> Thanks for the report Walt,
> 
> I fixed the issue with [Greek] in footnotes late last night. Could you 
> update to the latest version and try again? I'm not seeing the issues 
> you are using the latest version.
> 
> Also, for the time being until things are more complete, I'm only 
> bumping the version on tagged releases. To reference a build it would 
> be more helpful to provide the commit hash instead of the version.
> 
> ```
> By the way, the help for dp2ppgen indicates that --fndest an be
> "inline", but if one specifies that all the footnotes seem to
> disappear.
> ```
> 
> Inline footnotes have not been implemented yet, but I probably should 
> put a warning or something up when fndest==inline for the time being.
> 
> ```
> [Edit: I was hoping it would just leave them where encountered,
> though rejoining across page boundaries. That's the option I would
> ultimately like for working with the enhanced footnote processing
> in ppgen when footnotes are to occur at chapter- or book-end.
> Leave them where you found them, but emit the .fm directives at
> each chapter end, or at the book end. Then the PPer can adjust the
> .fm options as desired. The only time you'd need to move a
> footnote would be when the PPer wants them after paragraphs.]
> ```
> 
> Currently, dp2ppgen still handles footnotes according to the old 
> footnote ppgen implementation. Footnotes will work the way you're 
> describing once I've updated dp2ppgen.
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/davem2/dp2ppgen/issues/4#issuecomment-82567638.

How do I find the commit hash? In any case, unless you've made changes 
since I filed that report I'm using the latest level that Git 
Fetch/Merge gives me.

Thanks,
Walt

**Comment by davem2** [comment]
To view commit hashes on a local repo:
`git log`

To vew them on Github:
https://github.com/davem2/dp2ppgen/commits/master/dp2ppgen/dp2ppgen.py

Also, the latest commit is shown on the top right of the file list on the project home page.

Could you send/link me a copy of the input file you are using? I wasn't able to reproduce the issue with those snippets you gave (using commit cd95b202cc).

This is what I get for output (looks correct to me):

```
.fn 1
[Greek:

/*
pylla[**polla? P1] d' en metaichmiô
notos kylindei kymat' eureiês halos.
*/

]

The author is unknown.
.fn-
.fn 2
Homer, Od..xii. 209:

/*
[Greek: on men dê tode meixon epi kakon ê ote Kuklôps
eilei eni skêi glaphyrps kraterêphi biêphi.]
*/

Cicero means that they have no more reason to resist Cæsar than before,
for his claim to keep his province is not more unconstitutional than his
former claims. "We can't suppose that our object in yielding the old
demands was to arm him against ourselves!" Nisi forte is here practically
a negative; it introduces an impossible supposition.
.fn-
```

**Comment by wf49670** [comment]
That's the commit I'm using. Thanks for the info on finding the data.

Here's the source file: 
https://dl.dropboxusercontent.com/u/3229599/cicerov2-orig2.txt
And the output: 
https://dl.dropboxusercontent.com/u/3229599/cicerov2-2b-src.txt

Regards,
Walt

**Comment by wf49670** [comment]
Forgot to add: look at 182.png and 240.png.

```
 Walt
```

**Comment by davem2** [comment]
Thanks for the files Walt.

I ran https://dl.dropboxusercontent.com/u/3229599/cicerov2-orig2.txt using cd95b202cc14eb96
`dp2ppgen cicerov2-orig2.txt -pcfjse --fndest=chapterend`

Here is a copy of my output file:
https://dl.dropboxusercontent.com/u/59728548/cicerov2-orig2-out.txt

I see the issues on 182.png and 240.png where part of the [Footnote] markup is left behind, but the generated .fn statements don't have the cutoff issue. I get the same output I posted above. 

I'm not sure why we're getting different results with the .fn statements. Did you by chance install dp2ppgen using pip? It may be that the copy of dp2ppgen you are running is not the one you think. Maybe its a platform issue, though I'm not sure how. I'll try running on Windows and see if it makes a difference.
